go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/containerd/containerd
 
-go 1.21
+go 1.22
```

Comment:
Curious; was there a requirement to _disallow_ 1.21? versions in go.mod are a hard enforcement now)

Reply:
My thought was should we support building on an EOL version of golang. The api module is still at 1.21 so the importers wont be impacted. 

Is it causing any breakages?

Reply:
The API modules is only a small part; the containerd main module contains most of the code and has a much larger audience.

For sure, consumers _should_ update Go versions, but some of them may not be immediately; where possible, the go version in go.mod should follow same conventions as other dependencies in the module, and specify the minimum _required_ version, which may be lower than the version used by containerd itself to build binaries. 

e.g. see the `go.mod` for `golang.org/x/sys` (and other `golang.org/x/` libraries), which only recently raised the minimum to go1.18; https://github.com/golang/sys/blob/914deed708990c429d263121ee1ef42388e94ba4/go.mod#L1-L3

```
module golang.org/x/sys


go 1.18
```

Reply:
Should we revert the go.mod change then?

Reply:
> Should we revert the go.mod change then?

Yes, please!